developers.adoc

```diff
@@ -225,25 +225,37 @@ to `X.Y.Z` format (e.g. `0.9.0` for 0.9 release)
 
 === Python
 
-* Create a `~/.pypirc` file with this (https://github.com/pypa/setuptools/issues/941#issuecomment-279123202[source]):
+* Make sure that the needed Python packaging tools are installed and up-to-date:
+** Debian/Ubuntu/etc.: `sudo apt install python3-setuptools python3-wheel twine`
+** Other Linuxes: see https://packaging.python.org/guides/installing-using-linux-tools/[Python Packaging User Guide]
+** Everywhere else: use your system package manager, or `python3 -m pip install --upgrade setuptools twine wheel`
+* Pump version in `kaitaistruct.py`, seek `\\__version__ =`
+* Build distributions and upload them to PyPI:
 +
-[source,ini]
+[source,bash]
 ----
-[distutils]
-index-servers =
-    pypi
-
-[pypi]
-repository: https://pypi.python.org/pypi
-username: <login>
-password: <pass>
+# Delete any old previously built distributions.
+$ rm -r dist/
```

Comment:
Well, OK, but I'd rather leave this out completely. Doesn't it become to some development literacy 😃? I'd say it's somewhat native to clean the generated folders just to be on the safe side, to start with a clean slate before the release. And BTW, have you tested whether it's necessary for the procedure to work?

Yet one more thing: this will fail if the `dist/` doesn't exist, so there should be definitely a `-f` flag:

```suggestion
# Delete any old previously built distributions.
$ rm -rf dist/
```

And the comment is even more obvious I think...

Reply:
Sure, it's good practice to start with a clean environment when making a release, and if you remember to do that, great. I put this in the instructions as a reminder just in case, so that it's less likely that someone uploads junk files to PyPI by accident.

Especially because with some other languages/tools you don't necessarily have to manually clean up previous build artifacts before making a release. For example when using Gradle to publish a JVM library, it will upload only the jars for the current version (that were just built) to the Maven repo, and not any other random files that might still be under build/libs.

Regarding the comment - yeah, the way I worded it, it's a bit redundant. I suppose if anything I should have explained the problem with accidentally uploading old files by accident.